pyleros/leros.py

```diff
@@ -0,0 +1,59 @@
+from myhdl import instances, block, Signal, intbv, \
+                    always_comb
+
+from pyleros.types import alu_op_type, IM_BITS, DM_BITS
+from pyleros.codes import dlist
+
+from pyleros import fedec, execute
+
+
+@block
+def pyleros(clk, reset, filename=None):
+    """The main pyleros module. Instantiates both the fedec 
+    and execute modules corresponding to the two pipeline stages, 
+    and connects them with signals
+
+    Arguments (ports):
+        clk: IN Clock signal
+        reset: IN Async reset signal
+        
+    Parameters:
+        filename: The file/ list containing instructions
+            to load into instruction memory.
+
+    """
+
+    # Accumulator, OUT from exec, IN to fedec
+    acc = Signal(intbv(0)[16:])
+
+    # read data from DM, IN to fedec, OUT from exec
+    dm_data = Signal(intbv(0)[16:])
+
+    # Decoder control signals list
+    d = {}
+    for i in dlist:
+        d[str(i)] = Signal(bool(0))
+
+    d['op'] = Signal(alu_op_type.LD)
+    dec = [d[str(sig)] for sig in dlist]
```

Comment:
This will not be convertible, do you have any conversion tests yet?

The `dec` list of signals has mixed types, for a list of signals to be convertible each element needs to be the same type.   There is not restriction in simulation (modeling).